sympy/polys/rootoftools.py

```diff
@@ -774,13 +774,58 @@ def _get_roots(cls, method, poly, radicals):
             if x.name not in free_names:
                 poly = poly.xreplace({d: x})
                 break
+
+        # compute rational lift if real algebraic
+        has_alg_coeff = any(c.is_algebraic and not c.is_rational
+                            for c in poly.all_coeffs())
+        has_alg_coeff = has_alg_coeff and all(c.is_real
+                            for c in poly.all_coeffs())
```

Comment:
Check `is_algebraic` is not quite right because:
```python
In [1]: x = Symbol('x', integer=True)

In [2]: x
Out[2]: x

In [3]: x.is_algebraic
Out[3]: True
```
The check should really be `c.is_algebraic and c.is_number`. However I don't think we need to do this check because the `Poly` constructor already handles this.

Reply:
Ah good point